esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp

```diff
@@ -1603,34 +1603,55 @@ static int bindingCompareFunc(IInterface * const *_itm1, IInterface * const* _it
         return 0;
 }
 
+
+void getAllEspProcesses(StringArray& processes)
+{
+    VStringBuffer xpath("/Environment/Software");
+    Owned<IRemoteConnection> conn = querySDS().connect(xpath.str(), myProcessSession(), RTM_LOCK_READ , SDS_LOCK_TIMEOUT_DESDL);
+    if (!conn)
+        throw MakeStringException(-1, "Unable to connect to /Environment/Software dali path");
+    Owned<IPropertyTreeIterator> iter = conn->queryRoot()->getElements("EspProcess");
+    ForEach (*iter)
+    {
+        IPropertyTree &item = iter->query();
+        processes.append(item.queryProp("@name"));
+    }
+
+    processes.sortAscii();
+}
+
 bool CWsESDLConfigEx::onListESDLBindings(IEspContext &context, IEspListESDLBindingsRequest&req, IEspListESDLBindingsResponse &resp)
 {
     if (m_isDetachedFromDali)
         throw MakeStringException(-1, "Cannot list ESDL Bindings. ESP is currently detached from DALI.");
 
-    Owned<IPropertyTree> esdlBindings = m_esdlStore->getBindings();
-    if (esdlBindings.get() == nullptr)
-        return false;
-    Owned<IPropertyTreeIterator> iter = esdlBindings->getElements("Binding");
     double ver = context.getClientVersion();
+    Owned<IPropertyTree> esdlBindings = m_esdlStore->getBindings();
     IArrayOf<IEspESDLBinding> list;
-    ForEach(*iter)
+    if (esdlBindings.get() != nullptr)
     {
-        IPropertyTree &item = iter->query();
-        Owned<IEspESDLBinding> esdlbinding = createESDLBinding("","");
-        esdlbinding->setId(item.queryProp("@id"));
-        esdlbinding->setEspProcess(item.queryProp("@espprocess"));
-        const char* portstr = item.queryProp("@port");
-        if (portstr && *portstr)
-            esdlbinding->setPort(atoi(portstr));
-        else
-            esdlbinding->setPort(0);
-        esdlbinding->setEspBinding(item.queryProp("@espbinding"));
-        list.append(*esdlbinding.getClear());
+        Owned<IPropertyTreeIterator> iter = esdlBindings->getElements("Binding");
+        ForEach(*iter)
+        {
+            IPropertyTree &item = iter->query();
+            Owned<IEspESDLBinding> esdlbinding = createESDLBinding("","");
+            esdlbinding->setId(item.queryProp("@id"));
+            esdlbinding->setEspProcess(item.queryProp("@espprocess"));
+            const char* portstr = item.queryProp("@port");
+            if (portstr && *portstr)
+                esdlbinding->setPort(atoi(portstr));
+            else
+                esdlbinding->setPort(0);
+            esdlbinding->setEspBinding(item.queryProp("@espbinding"));
+            list.append(*esdlbinding.getClear());
+        }
+        list.sort(bindingCompareFunc);
     }
-    list.sort(bindingCompareFunc);
     if (ver >= 1.4)
     {
+        StringArray allProcNames;
```

Comment:
It's clever to create the logic based on the sorted array, but we would benefit from either sorting here, or at least renaming allProcNames to allProcNamesSortedAlpaha